homeassistant/components/roon/media_player.py

```diff
@@ -478,6 +508,110 @@ def play_media(self, media_type, media_id, **kwargs):
                 media_id,
             )
 
+    async def async_join(self, join_ids):
+        """Add another Roon player to this player's join group."""
+
+        zone_data = self._server.roonapi.zone_by_output_id(self._output_id)
+        if zone_data is None:
+            _LOGGER.error("No zone data for %s", self.name)
+            return
+
+        sync_available = {}
+        for zone in self._server.zones.values():
+            for output in zone["outputs"]:
+                if (
+                    zone["display_name"] != self.name
+                    and output["output_id"]
+                    in self.player_data["can_group_with_output_ids"]
+                    and zone["display_name"] not in sync_available.keys()
+                ):
+                    sync_available[zone["display_name"]] = output["output_id"]
+
+        names = []
+        for entity_id in join_ids:
+            name = self._server.roon_name(entity_id)
+            if name is None:
+                _LOGGER.error("No roon player found for %s", entity_id)
+                return
+            if name not in sync_available.keys():
+                _LOGGER.error(
+                    "Can't join player %s with %s because it's not in the join available list %s",
+                    name,
+                    self.name,
+                    list(sync_available.keys()),
+                )
+                return
+            names.append(name)
+
+        _LOGGER.info("Joining %s to %s", names, self.name)
+        self._server.roonapi.group_outputs(
```

Comment:
This is a sync method and should be called in the executor.
```suggestion
        await self.hass.async_add_executor_job(
            self._server.roonapi.group_outputs,
```